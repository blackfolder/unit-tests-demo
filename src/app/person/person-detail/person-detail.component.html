<div unitTestsDemoColorChanger>
    <h2 (click)="onHeaderClicked()" class="header">Say my name!</h2>

    <p>{{ firstName }} {{ lastName }}</p>
</div>