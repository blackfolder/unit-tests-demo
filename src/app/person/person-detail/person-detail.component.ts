import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
    selector: 'unit-tests-demo-person-detail',
    template: `,
    <div unitTestsDemoColorChanger>
        <h2 (click)="onHeaderClicked()" class="header">Say my name!</h2>
        <p>{{ firstName }} {{ lastName }}</p>
    </div>`,
    styleUrls: ['./person-detail.component.scss'],
})
export class PersonDetailComponent {
    @Input() firstName: string | undefined = 'No name';
    @Input() lastName: string | undefined  = 'No last name';
    
    @Output() headerCliked = new EventEmitter<string>();

    onHeaderClicked(): void {
        this.headerCliked.emit('Header clicked!');
    }
}
