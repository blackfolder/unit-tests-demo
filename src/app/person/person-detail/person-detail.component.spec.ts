import { DebugElement } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { PersonDetailComponent } from './person-detail.component';

describe('PersonDetailComponent', () => {
    let component: PersonDetailComponent;
    let fixture: ComponentFixture<PersonDetailComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            declarations: [PersonDetailComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(PersonDetailComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should trigger output event when header clicked', () => { // Why we don't need waitForAsync?
        const debugElement: DebugElement = fixture.debugElement;
        const header = debugElement.query(By.css('h2'));
        let onClickMessage;
        component.headerCliked.subscribe((message) => onClickMessage = message);

        header.triggerEventHandler('click');

        expect(onClickMessage).toBe('Header clicked!');
    });

    it('should render correctly - snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });
});
