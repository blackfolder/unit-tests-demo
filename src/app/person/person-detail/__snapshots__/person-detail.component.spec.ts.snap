// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetailComponent should render correctly - snapshot 1`] = `
<unit-tests-demo-person-detail
  __ngContext__={[Function Number]}
  firstName={[Function String]}
  headerCliked={[Function EventEmitter_]}
  lastName={[Function String]}
>
  , <div
    unittestsdemocolorchanger=""
  >
    <h2
      class="header"
    >
      Say my name!
    </h2>
    <p>
      No name No last name
    </p>
  </div>
</unit-tests-demo-person-detail>
`;
